weave/initialization/pil_image_thread_safety.py

```diff
@@ -0,0 +1,52 @@
+"""
+This file exposes a function that makes the PIL ImageFile thread-safe: `make_image_file_thread_safe`.
+
+There is a discussion here: https://github.com/python-pillow/Pillow/issues/4848#issuecomment-671339193 in which
+the author claims that the Pillow library is thread-safe. However, my reasoning leads me to a different conclusion.
+
+Specifically, the `ImageFile.load` method is not thread-safe. This is because `load` will both close and delete
+an open file handler as well as modify properties of the ImageFile object (namely the `im` property which contains
+the underlying image data). Inside of Weave we use threads to parallelize work which may involve Images. This bug
+has presented itself not only in our own persistence layer, but also in user code where they are consuming loaded
+images across threads.
+
+We call `make_image_file_thread_safe` in the `__init__.py` file to ensure that the ImageFile class is thread-safe.
+"""
+
+import threading
+from functools import wraps
+
+_patch_lock = threading.Lock()
+_patched = False
+
+
+def make_image_file_thread_safe() -> None:
+    global _patched
+    with _patch_lock:
+        if _patched:
+            return
+        _patched = True
```

Comment:
you should move this after the try except block, probably as:
```py
try:
    _make_image_file_thread_safe()
except Exception as e:
    print(f"Failed to patch PIL.ImageFile.ImageFile: {e}.")
else:
    _patched = True
```

Reply:
Unless `_patched` here is meant to be "attempted patch"

Reply:
yup - fixing